src/vs/workbench/contrib/accessibility/browser/accessibleViewContributions.ts

```diff
@@ -349,3 +352,67 @@ export class InlineCompletionsAccessibleViewContribution extends Disposable {
 	}
 }
 
+export class ExtensionAccessibilityHelpDialogContribution extends Disposable {
+	static ID = 'extensionAccessibilityHelpDialogContribution';
+	private _viewHelpDialogMap = this._register(new DisposableMap<string, IDisposable>());
+	constructor(@IKeybindingService keybindingService: IKeybindingService) {
+		super();
+		this._register(Registry.as<IViewsRegistry>(Extensions.ViewsRegistry).onViewsRegistered(e => {
+			for (const view of e) {
+				for (const viewDescriptor of view.views) {
+					if (viewDescriptor.accessibilityHelpContent) {
+						this._viewHelpDialogMap.set(viewDescriptor.id, registerAccessibilityHelpAction(keybindingService, viewDescriptor));
+					}
+				}
+			}
+		}));
+		this._register(Registry.as<IViewsRegistry>(Extensions.ViewsRegistry).onViewsDeregistered(e => {
+			for (const viewDescriptor of e.views) {
+				if (viewDescriptor.accessibilityHelpContent) {
+					this._viewHelpDialogMap.get(viewDescriptor.id)?.dispose();
+				}
+			}
+		}));
+	}
+}
+
+function registerAccessibilityHelpAction(keybindingService: IKeybindingService, viewDescriptor: IViewDescriptor): IDisposable {
+	const disposableStore = new DisposableStore();
+	const helpContent = resolveExtensionHelpContent(keybindingService, viewDescriptor.accessibilityHelpContent);
+	if (!helpContent) {
+		throw new Error('No help content for view');
+	}
+	disposableStore.add(AccessibilityHelpAction.addImplementation(95, viewDescriptor.id, accessor => {
+		const accessibleViewService = accessor.get(IAccessibleViewService);
+		const viewsService = accessor.get(IViewsService);
+		accessibleViewService.show(new ExtensionContentProvider(
+			viewDescriptor.id,
+			{ type: AccessibleViewType.Help },
+			() => helpContent.value,
+			() => viewsService.openView(viewDescriptor.id)
+		));
+		return true;
+	}, FocusedViewContext.isEqualTo(viewDescriptor.id)));
+	disposableStore.add(keybindingService.onDidUpdateKeybindings(() => {
+		disposableStore.clear();
+		disposableStore.add(registerAccessibilityHelpAction(keybindingService, viewDescriptor));
+	}));
+	return disposableStore;
+}
+
+function resolveExtensionHelpContent(keybindingService: IKeybindingService, content?: MarkdownString): MarkdownString | undefined {
+	if (!content) {
+		return;
+	}
+	let resolvedContent = typeof content === 'string' ? content : content.value;
+	const matches = resolvedContent.matchAll(/\(kb\(command:(?<commandId>.*)\)\)/gm);
```

Comment:
Instead of inventing a new syntax, can we re-use the markdown link syntax? This could either be a normal markdown link:

```
[optional extra text](command:commandId....)

or maybe:

[optional extra text](keybinding:commandId)

```

Or use the auto link syntax:

```
<command:commandId....>

<vscode-keybinding:commandId....>
```

The advantage is that we get the parsing essentially for free and it will be much more reliable than string replacements

Reply:
Yes, makes sense. Far preferable if we don't have to do extra parsing. I was not aware of the auto link syntax.

Reply:
I also wonder if we should have syntax which allows someone to just provide the command id and have it render [Command Title] (keybinding).

When I tested contributing this to a view, it got redundant quickly:
```
"accessibilityHelpContent": "Helpful commands include: \n - Refresh Issue List (kb(command:issue.refresh))\n-GitHub Pull Requests: Focus on Issues View (kb(command:issues:github.focus)) \n-GitHub Pull Requests: Focus on Pull Requests View (kb(command:pr:github.focus))\n-GitHub Issues: Copy GitHub Permalink (kb(command:issue.copyGithubPermalink))\n-GitHub Issues: Create an Issue (kb(command:issue.createIssue))\n-GitHub Pull Requests: Create Pull Request (kb(command:pr.create))\n-GitHub Pull Requests: Start Review (kb(command:pr.startReview))\n-GitHub Pull Requests: Mark Pull Request Ready For Review (kb(command:pr.readyForReview))\n-GitHub Pull Requests: Approve (kb(command:review.approve))"
```

Reply:
I'm not sure where the auto-link syntax is used 🤔

Reply:
I am changing it to `(keybinding:commandId)`.

Reply:
I have also added support for configuring the keybinding when there isn't one.